### ISSUE TITLE
Type arguments from unannotated assemblies are treated as not nullable

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#27967]
A warning is reported for assignment to `b.F[0]` even though the type of `F` is from an unannotated assembly.
```c#
// csc.exe /t:library /langversion:7 a.cs
public class A<T>
{
    public T F;
}
public class B : A<object[]>
{
}
```
```c#
// csc.exe /t:library /r:a.dll b.cs
class C
{
    static void Main()
    {
        var b = new B();
        b.F[0] = null;
    }
}
```
